src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/rca/jvmsizing/OldGenReclamationRca.java

```diff
@@ -0,0 +1,147 @@
+    this.minOldGenSlidingWindow = new MinOldGenSlidingWindow(1, TimeUnit.MINUTES);
+    this.gcEventsSlidingWindow = new SlidingWindow<>(1, TimeUnit.MINUTES);
+    this.prevContext = new ResourceContext(State.UNKNOWN);
+    this.prevSummary = null;
+  }
+
+  @Override
+  public void generateFlowUnitListFromWire(FlowUnitOperationArgWrapper args) {
+    throw new UnsupportedOperationException("generateFlowUnitListFromWire should not be called "
+        + "for node-local rca: " + args.getNode().name());
+  }
+
+  @Override
+  public ResourceFlowUnit<HotResourceSummary> operate() {
+    samples++;
+    double oldGenMax = getOldGenValueForMetric(heapMax);
+    double oldGenUsed = getOldGenValueForMetric(heapUsed);
+    double gcEvents = getGcEvents();
+    long currTime = System.currentTimeMillis();
+    minOldGenSlidingWindow.next(new SlidingWindowData(currTime, oldGenUsed));
+    gcEventsSlidingWindow.next(new SlidingWindowData(currTime, gcEvents));
+
+    if (samples == rcaPeriod) {
+      samples = 0;
+      double events = gcEventsSlidingWindow.readSum();
+      if (events >= 1) {
+        double threshold = targetHeapUtilizationAfterGc / 100d * oldGenMax;
+        HotResourceSummary summary = null;
+        ResourceContext context = null;
+        if (minOldGenSlidingWindow.readMin() > threshold) {
+          summary = new HotResourceSummary(ResourceUtil.FULL_GC_EFFECTIVENESS,
+              targetHeapUtilizationAfterGc, minOldGenSlidingWindow.readMin(), 60);
+          context = new ResourceContext(State.UNHEALTHY);
+
+          return new ResourceFlowUnit<>(currTime, context, summary);
+        } else {
+          summary = new HotResourceSummary(ResourceUtil.FULL_GC_EFFECTIVENESS,
+              targetHeapUtilizationAfterGc, minOldGenSlidingWindow.readMin(), 60);
+          context = new ResourceContext(State.HEALTHY);
+        }
+
+        prevSummary = summary;
+        prevContext = context;
+
+        return new ResourceFlowUnit<>(currTime, context, summary);
+      }
+    }
+
+    return new ResourceFlowUnit<>(currTime, prevContext, prevSummary);
+  }
+
+  private double getGcEvents() {
+    List<MetricFlowUnit> gcEventMetricFlowUnits = gcEvent.getFlowUnits();
+    double metricValue = 0d;
+    for (final MetricFlowUnit gcEventMetricFlowUnit : gcEventMetricFlowUnits) {
+      if (gcEventMetricFlowUnit.isEmpty()) {
+        continue;
+      }
+
+      double ret = SQLParsingUtil.readDataFromSqlResult(gcEventMetricFlowUnit.getData(),
+          MEM_TYPE.getField(),
+          TOT_FULL_GC.toString(), MetricsDB.MAX);
+      if (!Double.isNaN(ret)) {
+        metricValue = ret;
+      }
+    }
+
+    return metricValue;
+  }
+
+  private double getOldGenValueForMetric(Metric heapMetric) {
+    List<MetricFlowUnit> heapMetricFlowUnits = heapMetric.getFlowUnits();
+    double metricValue = 0d;
+    for (final MetricFlowUnit heapMetricFlowUnit : heapMetricFlowUnits) {
+      if (heapMetricFlowUnit.isEmpty()) {
+        continue;
+      }
```

Comment:
These two functions are really useful. Can you move them into a MetricUtil class so that they can be reused?

Reply:
Instead of putting them in a static class, I've reduced duplication by abstracting these old gen related methods into an abstract class OldGenRca from which other old gen profiling RCAs extend.